vaadin-grid-pro-flow-integration-tests/src/main/java/com/vaadin/flow/component/gridpro/examples/MainView.java

```diff
@@ -85,20 +111,26 @@ protected void createBeanGridWithEditColumns() {
         add(beanGrid);
     }
 
-    private static List<Person> createItems() {
+    private static List<Person> createItems(List<City> cityList) {
         Random random = new Random(0);
         return IntStream.range(1, 500)
-                .mapToObj(index -> createPerson(index, random))
+                .mapToObj(index -> createPerson(index, random, cityList))
                 .collect(Collectors.toList());
     }
 
-    private static Person createPerson(int index, Random random) {
+    private static Person createPerson(int index, Random random,
+                                       List<City> cityList) {
         Person person = new Person();
         person.setId(index);
         person.setEmail("person" + index + "@vaadin.com");
         person.setName("Person " + index);
         person.setAge(13 + random.nextInt(50));
 
+        City city = new City("City" + index, person);
```

Comment:
Even for a test view, it's too hacky to create cities and add them to the list as a side effect of `createPerson`.

Reply:
Updated. Created `mapLists`